src/connect.js

```diff
@@ -4,7 +4,9 @@ import { core, object } from 'metal';
 import JSXComponent from 'metal-jsx';
 
 const defaultMapStateToProps = () => ({});
-const defaultMapDispatchToProps = dispatch => ({ dispatch });
+const defaultMapDispatchToProps = dispatch => ({
+	dispatch
+});
 const defaultMergeProps = (stateProps, dispatchProps, parentProps) => {
 	return object.mixin({}, stateProps, dispatchProps, parentProps);
```

Comment:
I believe that this still needs to be addressed, as now that you've fixed the order of objects that are being passed to this function we need to fix the order in which they're merged. This basically needs to be `object.mixin({}, parentProps, stateProps, dispatchProps);`.

It'd also be good to add a test case to make sure there aren't any regressions for this. An example of a test would be a component that receives a prop named `foo` and then defines a prop via `mapStoreStateToProps` that is also named `foo`. The value of the last one should be the one that the component gets in the end, not the original prop.